ipatests/test_integration/base.py

```diff
@@ -112,3 +114,49 @@ def uninstall(cls, mh):
             tasks.uninstall_client(client)
         if cls.fips_mode:
             cls.disable_fips_mode()
+
+
+@ordered
+@pytest.mark.usefixtures('mh')
+@pytest.mark.usefixtures('integration_logs')
+class MultiDomainIntegrationTest(IntegrationTest):
+    num_trusted_domains = 1
+    num_trusted_replicas = 0
+    num_trusted_clients = 0
+
+    @classmethod
+    def get_domains(cls):
+        return super(MultiDomainIntegrationTest, cls
+                     ).get_domains() + cls.trusted_domains
+
+    @classmethod
+    def install(cls, mh):
+        super(MultiDomainIntegrationTest, cls).install(mh)
+        extra_args = []
+        if cls.topology is None:
+            return
+        else:
+            if cls.token_password:
+                extra_args.extend(('--token-password', cls.token_password,))
+            tasks.install_topo(cls.topology,
+                               cls.trusted_master, cls.trusted_replicas,
+                               cls.trusted_clients, 1,
+                               random_serial=cls.random_serial,
+                               extra_args=extra_args,)
```

Comment:
In order to prepare the trust setup, the install method can also configure DNSSEC validation:
```
        # Now enable dnssec on the zones
        cls.master.run_command([
            "ipa-dns-install",
            "--dnssec-master",
            "--forwarder", cls.master.config.dns_forwarder,
            "-U",
        ])
        cls.master.run_command([
            "ipa", "dnszone-mod", cls.master.domain.name,
            "--dnssec=True"
        ])
        cls.trusted_master.run_command([
            "ipa-dns-install",
            "--dnssec-master",
            "--forwarder", cls.trusted_master.config.dns_forwarder,
            "-U",
        ])
        cls.trusted_master.run_command([
            "ipa", "dnszone-mod", cls.trusted_master.domain.name,
            "--dnssec=True"
        ])
```